workers/unity/Assets/Playground/Editor/DisconnectOnCompilation.cs

```diff
@@ -0,0 +1,36 @@
+using UnityEditor;
+using UnityEditor.Compilation;
+using UnityEngine;
+
+namespace Playground.Editor
+{
+    public class DisconnectOnCompilation
+    {
+        private static bool stopped;
+
+        [RuntimeInitializeOnLoadMethod]
+        private static void Init()
+        {
+            CompilationPipeline.assemblyCompilationStarted += OnCompilationStarted;
+            CompilationPipeline.assemblyCompilationFinished += OnCompilationFinished;
+        }
+
+        private static void OnCompilationStarted(string obj)
+        {
+            if (EditorApplication.isPlaying)
+            {
+                EditorApplication.isPlaying = false;
+                stopped = true;
+            }
+        }
+
+        private static void OnCompilationFinished(string arg1, CompilerMessage[] arg2)
```

Comment:
What if compilation has failed?

Reply:
It will try to start the game and because there are compiler errors it won't.

Reply:
I'd prefer it to not even try to start if compilation failed

Reply:
Yup, changed that it doesn't try to restart if there are any error messages. But will try if its only warnings.